engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobYieldProcessor.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnBehaviors;
+import io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnJobActivationBehavior;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.JobState;
+import io.camunda.zeebe.engine.state.immutable.JobState.State;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+
+public final class JobYieldProcessor implements TypedRecordProcessor<JobRecord> {
+
+  public static final String NOT_ACTIVATED_JOB_MESSAGE =
+      "Expected to time out activated job with key '%d', but %s";
+  private final JobState jobState;
+  private final BpmnJobActivationBehavior jobActivationBehavior;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+
+  public JobYieldProcessor(
+      final ProcessingState state, final BpmnBehaviors bpmnBehaviors, final Writers writers) {
+    jobState = state.getJobState();
+    jobActivationBehavior = bpmnBehaviors.jobActivationBehavior();
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<JobRecord> record) {
+    final long key = record.getKey();
+    final JobState.State state = jobState.getState(key);
+    if (state == State.ACTIVATED) {
+      final JobRecord yieldedJob = jobState.getJob(key);
+
+      stateWriter.appendFollowUpEvent(key, JobIntent.YIELDED, yieldedJob);
+      jobActivationBehavior.notifyJobAvailable(yieldedJob);
```

Comment:
:wrench: I wonder if we should add a code comment here that the job notification is done through a side-effect, as it might not be clear with the method name, i.e. it might seem that we do it immediately.

Reply:
@koevskinikola I think, this is one will become invalid with the changes in JobStreamer, right?

Reply:
@berkaycanbc it's still valid. The JobStreamer API will be changed so that we don't need to pass the `ErrorHandler` with every job push, but only pass a single instance when the `JobStreamer` is instantiated. 

In the `ErrorHandler` implementation, we will still be using the job `YIELD` command, and as we confirmed in the meeting yesterday if a job is yielded, it falls back under the long polling mechanism. So we still need to do a notification.